ezmlm-browse-conf.1
.TH ezmlm-browse-conf 1
.SH NAME
setup \- what it does
.SH SYNOPSIS
.B ezmlm-browse-conf
[
.B \-\-apache
] [
.B \-\-setuid
]
.I DESTPATH TARGET.cgi [ LISTPATH HOSTNAME ]
.SH DESCRIPTION
This program sets up the necessary links and programs to use
.B ezmlm-browse
on a web server.
.P
.I DESTPATH
is the path into which the files should be installed.  This is
necessarily a directory which the web server can access.
.I TARGET.cgi
is the filename for the CGI script in the target directory.  This will
frequently be
.I index.cgi
if there is nothing else in the directory.
.P
If
.I LISTPATH
and
.I HOSTNAME
are present on the command line, then a sample configuration file is
written to the destination directory.
.I LISTPATH
is the full path to the parent directory to the mailing list archives.
.I HOSTNAME
is the domain name on which the mailing lists receive messages.
.SH OPTIONS
.TP
.B \-\-apache
Create an Apache
.B .htaccess
configuration file in the destination.
.TP
.B \-\-setuid
Install a setuid CGI wrapper in the destination (instead of the usual
startup script) in order to allow
.B ezmlm-browse
to read the archive files.  This is useful if the archive files are
owned by a different user than the web server and Apache suexec (or
equivalent) is not in use.


NEWS
-------------------------------------------------------------------------------
Changes in version 0.20

- Ezmlm-browse now properly converts all messages to UTF-8 on output.

- Switched configuration to use a config.ini file.

- Add an ezmlm-browse-conf script to simplify installation.

- Fixed numbering of attachments.

- Fixed handling of messages with no date stamp.

- Default templates now show attached images inline.

- Downloading of images is controlled separately from downloading the
  whole raw message.

Development of this version has been sponsored by FutureQuest, Inc.
ossi@FutureQuest.net  http://www.FutureQuest.net/
-------------------------------------------------------------------------------
Changes in version 0.13

- Added patch from John Simpson to allow masking email addresses to be
  turned off.

- Fixed handling archives with invalid month numbers.

- Fixed to work with Python 2.5.

Development of this version has been sponsored by FutureQuest, Inc.
ossi@FutureQuest.net  http://www.FutureQuest.net/
-------------------------------------------------------------------------------
Changes in version 0.12

- Moved all static files into a separate "files" directory.

- Fixed the distribution tarball to actuall include all source files.

- Fixed the previous "fix" to bogus month thread index filenames.

- Added a feed icon to the lists page from http://www.feedicons.com/

Note: The file move above will likely cause breakage on upgrades.  See
the UPGRADE file for more information.

Development of this version has been sponsored by FutureQuest, Inc.
ossi@FutureQuest.net  http://www.FutureQuest.net/
-------------------------------------------------------------------------------
Changes in version 0.11

- Fixed problem in handling of missing articles when searching, and
  missing month thread indexes.

- Fixed a problem with bogus month thread index filenames.

- Fixed several build issues, including problems finding the python
  program.

Development of this version has been sponsored by FutureQuest, Inc.
ossi@FutureQuest.net  http://www.FutureQuest.net/
-------------------------------------------------------------------------------
Changes in version 0.10

- Added support for generation of syndication feeds.

- Added support for reversing the order of the "by Date" message list.

- Added support for blocking or allowing for downloading raw messages.

- Added CSSPREFIX configuration item, to allow for easy configuration of
  where CSS files are located.  Thanks Roger Merchberger.

- Added support for word-wrapping message bodies (defaulting to off).
  Idea from Roger Merchberger.

Development of this version has been sponsored by FutureQuest, Inc.
ossi@FutureQuest.net  http://www.FutureQuest.net/
-------------------------------------------------------------------------------
Changes in version 0.9

- Two bugfixes in the handling of month numbers from "Central Linux
  Experts And Newbies" via Gerrit Pape.
- Some HTML template bugfixes.
-------------------------------------------------------------------------------
Changes in version 0.8

- Rewrote the front-end from scratch, separating content generation from
  the HTML templates.  This means less flexability in the templates, but
  faster code and easier maintenance and templating.
- Now has author indices, paging, more links, time zone management,
  style configuration, and other features I've forgotten.
- The default template is marked up with CSS classes in many places, and
  three sample style sheets are included.
-------------------------------------------------------------------------------
Changes in version 0.6

- Fixed a bug in EzmlmArchive.month.
- Added new built-in functions for producing relative links in the
  templates: relink and cmdlink.
- Rewrote the templates to merge all the common header and footer data
  into a single file.
- Added a parse tree optimizer to the template system which removes body
  elements consisting of a single node.
- Pre-compile all template expressions before use to speed up looped
  execution.
-------------------------------------------------------------------------------
Changes in version 0.5

- Totally rebuild the template system, moving most of the actual
  page-specific code into the template pages themselves.
-------------------------------------------------------------------------------
Changes in version 0.2

- Added proper targets to build the compiled Python bytecode files.

- Replaced the Python main routine with a C wrapper that is suitable for
  installations where setuid operation is required.
-------------------------------------------------------------------------------


ezmlm-browse-conf.py
import getopt
import os
import sys

argv0 = sys.argv[0]

try:
	opts,args = getopt.getopt(sys.argv[1:], '',
							  [ 'apache', 'setuid' ])
	opts = dict(opts)
except getopt.GetoptError, x:
	print '%s: Invalid option: %s' % (argv0, x)
	sys.exit(1)

if len(args) == 2:
	destbase,target = args
	listpath = None
	hostname = None
elif len(args) == 4:
	destbase,target,listpath,hostname = args
	if listpath[0] != '/':
		print '%s: LISTPATH should start with a "/".' % argv0
		sys.exit(1)
else:
	print 'usage: %s [options] DESTPATH TARGET.cgi [ LISTPATH HOSTNAME ]' % argv0
	print '  Options:'
	print '  --apache -- Create a .htaccess file for Apache'
	print '  --setuid -- Copy in a setuid wrapper program'
	sys.exit(1)

srcbase = os.getcwd()

def destpath(name):
	return os.path.join(destbase, name)

def srcpath(name):
	return os.path.join(srcbase, name)

if not os.path.isdir(destbase):
	os.mkdir(destbase, 0777)

try:
	os.unlink(destpath(target))
except:
	pass

if '--setuid' in opts:
	if os.getuid() == 0 or os.getgid() == 0:
		print '"%s: NEVER use the --setuid option as root' % argv0
		sys.exit(1)
	wrapper = open(srcpath('wrapper')).read()
	f = open(destpath(target), 'w')
	f.write(wrapper)
	f.close()
	os.chmod(destpath(target), 06755)
else:
	os.symlink(srcpath('browse.cgi'), destpath(target))

try:
	os.unlink(destpath('files'))
except:
	pass
os.symlink(srcpath('files'), destpath('files'))

if listpath:
	f = open(destpath('ezmlm-browse.ini'), 'w')
	f.write('''# Automatically generated configuration file
# All of the settings below
[globals]
allowraw = 0
mask_emails = 1
basedir = %(listpath)s
basehost = %(hostname)s

[defaults]
style = browse.css
perpage = 20
msgsperpage = 10
filesprefix = files/
wrapwidth = 0
feedtype = atom
feedmsgs = 10
datesort = ascending

[styles]
browse.css = Default
purple.css = Purple
greenterm.css = Terminal

[charsets]

# Replace this with suitable entries for your list.  listdir, listemail,
# and listsub are all automatically generated from the archive name and
# basedir and basehost above.

[archive:sample]
listdesc = Sample mailing list description
listdir = %(listpath)s/sample
listemail = sample@%(hostname)s
listsub = sample-subscribe@%(hostname)s
''' % locals())
	f.close()

if '--apache' in opts:
	f = open(destpath('.htaccess'), 'w')
	f.write('DirectoryIndex %s\n'
			'AddHandler cgi-script .cgi\n'
			'Options +ExecCGI\n' % target)
	f.close()
